kafka/signed_blinded_token_issuer_handler.go

```diff
@@ -76,47 +98,107 @@ func SignedBlindedTokenIssuerHandler(
 			}
 			blindedTokens = append(blindedTokens, &blindedToken)
 		}
-		// @TODO: If one token fails they will all fail. Assess this behavior
-		signedTokens, dleqProof, err := btd.ApproveTokens(blindedTokens, issuer.SigningKey)
-		if err != nil {
-			logger.Error().
-				Err(fmt.Errorf("request %s: could not approve new tokens: %w",
-					blindedTokenRequestSet.Request_id, err)).
-				Msg("signed blinded token issuer handler")
-			blindedTokenResults = append(blindedTokenResults, avroSchema.SigningResult{
-				Signed_tokens:     nil,
-				Issuer_public_key: "",
-				Status:            issuerError,
-				Associated_data:   request.Associated_data,
-			})
-			continue
-		}
-		marshaledDLEQProof, err := dleqProof.MarshalText()
-		if err != nil {
-			return fmt.Errorf("request %s: could not marshal dleq proof: %w", blindedTokenRequestSet.Request_id, err)
-		}
-		var marshaledTokens []string
-		for _, token := range signedTokens {
-			marshaledToken, err := token.MarshalText()
+		// if the issuer is time aware, we need to approve tokens
+		if issuer.Version == 3 && issuer.Buffer > 1 {
+			// number of tokens per signing key
+			var numT = len(request.Blinded_tokens) / (issuer.Buffer + issuer.Overlap)
+			// sign tokens with all the keys in buffer+overlap
+			for i := issuer.Buffer + issuer.Overlap; i > 0; i-- {
+				var signingKey *crypto.SigningKey
+				if len(issuer.Keys) > i {
+					signingKey = issuer.Keys[len(issuer.Keys)-i].SigningKey
+				}
+				// @TODO: If one token fails they will all fail. Assess this behavior
+				signedTokens, dleqProof, err := btd.ApproveTokens(blindedTokens[(i-numT):i], signingKey)
+				if err != nil {
+					logger.Error().
+						Err(fmt.Errorf("request %s: could not approve new tokens: %w",
+							blindedTokenRequestSet.Request_id, err)).
+						Msg("signed blinded token issuer handler")
+					blindedTokenResults = append(blindedTokenResults, avroSchema.SigningResult{
+						Signed_tokens:     nil,
+						Issuer_public_key: "",
+						Status:            issuerError,
+						Associated_data:   request.Associated_data,
+					})
+					continue
```

Comment:
I think we might have to break out of this loop and then out of the outer loop to avoid adding multiple `avroSchema.SigningResult` once we error.